src/ChainSafe.Gaming/Web3/Core/LifecycleHandler.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace ChainSafe.Gaming.Web3.Core
+{
+    /// <summary>
+    /// Handles the lifecycle of <see cref="ILifecycleParticipant"/> instances.
+    /// </summary>
+    public class LifecycleHandler
+    {
+        private readonly ILifecycleParticipant[] lifecycleParticipants;
+
+        public LifecycleHandler(IEnumerable<ILifecycleParticipant> lifecycleParticipants)
+        {
+            // Arrange execution based on ExecutionOrder Attribute priority.
+            this.lifecycleParticipants = lifecycleParticipants
+                .OrderBy(p => p.GetType().GetCustomAttribute<ExecutionOrderAttribute>()?.Order ?? 0)
```

Comment:
Yeah instead of a reflection why we didn't use the field? Since we do support default interface implementation in the core solution.

Reply:
Interfaces can't hold data by default just implementations/logic, so every concrete implementation will have to implement the `ExecutionOrder` property

Reply:
![image](https://github.com/user-attachments/assets/b2a04694-2e10-4612-abf1-fc4a5de0745c)
But you can do this.

Reply:
And if you need to override it you can do that in the implementator later on 
![image](https://github.com/user-attachments/assets/2a7983a6-80de-4f56-87fe-0cb024e6292a)

Reply:
I had no idea, C# reversing entropy out here xD

Reply:
I'll redo it

Reply:
done